pkg/ccloudv2/connect_custom_plugin.go

```diff
@@ -0,0 +1,94 @@
+package ccloudv2
+
+import (
+	"context"
+	"net/http"
+
+	connectcustompluginv1 "github.com/confluentinc/ccloud-sdk-go-v2/connect-custom-plugin/v1"
+
+	"github.com/confluentinc/cli/v3/pkg/errors"
+)
+
+func newConnectCustomPluginClient(url, userAgent string, unsafeTrace bool) *connectcustompluginv1.APIClient {
+	cfg := connectcustompluginv1.NewConfiguration()
+	cfg.Debug = unsafeTrace
+	cfg.HTTPClient = NewRetryableHttpClient(unsafeTrace)
+	cfg.Servers = connectcustompluginv1.ServerConfigurations{{URL: url}}
+	cfg.UserAgent = userAgent
+
+	return connectcustompluginv1.NewAPIClient(cfg)
+}
+
+func (c *Client) connectCustomPluginApiContext() context.Context {
+	return context.WithValue(context.Background(), connectcustompluginv1.ContextAccessToken, c.AuthToken)
+}
+
+func (c *Client) GetPresignedUrl(extension string) (connectcustompluginv1.ConnectV1PresignedUrl, error) {
+	request := *connectcustompluginv1.NewConnectV1PresignedUrlRequest()
+	request.SetContentFormat(extension)
+	resp, httpResp, err := c.CcpClient.PresignedUrlsConnectV1Api.PresignedUploadUrlConnectV1PresignedUrl(c.connectCustomPluginApiContext()).ConnectV1PresignedUrlRequest(request).Execute()
+	return resp, errors.CatchCCloudV2Error(err, httpResp)
+}
+
+func (c *Client) CreateCustomPlugin(displayName, description, documentationLink, connectorClass, connectorType, uploadId string, sensitiveProperties []string) (connectcustompluginv1.ConnectV1CustomConnectorPlugin, error) {
+	createCustomPluginRequest := connectcustompluginv1.NewConnectV1CustomConnectorPlugin()
+	createCustomPluginRequest.SetDisplayName(displayName)
+	createCustomPluginRequest.SetDescription(description)
+	createCustomPluginRequest.SetDocumentationLink(documentationLink)
+	createCustomPluginRequest.SetConnectorClass(connectorClass)
+	createCustomPluginRequest.SetConnectorType(connectorType)
```

Comment:
Can you move this logic outside of this function? Please imitate the style of other similar functions in other files in this package. We'd like to generate this code in the near future and it would help if it was consistent!